google-oauth-client-jetty/src/main/java/com/google/api/client/extensions/jetty/auth/oauth2/LocalServerReceiver.java

```diff
@@ -128,22 +119,16 @@ public String getRedirectUri() throws IOException {
 
   @Override
   public String waitForCode() throws IOException {
-    lock.lock();
-    try {
-      while (code == null && error == null) {
-        gotAuthorizationResponse.awaitUninterruptibly();
-      }
-      if (error != null) {
-        throw new IOException("User authorization failed (" + error + ")");
-      }
-      return code;
-    } finally {
-      lock.unlock();
+    waitUnlessSignaled.acquireUninterruptibly();
+    if (error != null) {
```

Comment:
I think returning null makes sense for `stop()`: `error` comes from HTTP request parameter, and stopping a server doesn't really have to throw an IOException. I'll add a JavaDoc for `waitForCode()` noting that `null` will be returned if `stop()` is called.